keras_core/layers/preprocessing/random_crop.py

```diff
@@ -75,7 +75,33 @@ def __init__(self, height, width, seed=None, name=None, **kwargs):
     def call(self, inputs, training=True):
         if not isinstance(inputs, (tf.Tensor, np.ndarray, list, tuple)):
             inputs = tf.convert_to_tensor(backend.convert_to_numpy(inputs))
-        outputs = self.layer.call(inputs, training=training)
+
+        input_shape = tf.shape(inputs)
```

Comment:
We seem to be pretty close to achieving backend-agnosticism -- can you try using `ops` instead of `tf` everywhere?